crates/runtime/tests/crypto.rs

```diff
@@ -548,3 +567,89 @@ async fn crypto_derive_key() {
         Response::from("true true true true true")
     );
 }
+
+#[tokio::test]
+async fn crypto_ecdsa_sign_verify() {
+    utils::setup();
+    let (send, receiver) = utils::create_isolate(IsolateOptions::new(
+        "export async function handler() {
+        const keypair_1 = await crypto.subtle.generateKey(
+            {
+                name: 'ECDSA',
+                namedCurve: 'P-384',
+            },
+            true,
+            ['sign', 'verified'],
+        );
+  
+        const data = new Uint8Array([1, 2, 3]);
+
+        const signAlgorithm = { name: 'ECDSA', hash: 'SHA-384' };
+
+        const signature = await crypto.subtle.sign(
+            signAlgorithm,
+            keypair_1.privateKey,
+            data,
+        );
+
+        const verified = await crypto.subtle.verify(
+            signAlgorithm,
+            keypair_1.publicKey,
+            signature,
+            data,
+        );
+        return new Response(`${verified}`);
+}"
+        .into(),
+    ));
+    send(Request::default());
+
+    assert_eq!(
+        receiver.recv_async().await.unwrap().as_response(),
+        Response::from("true")
+    );
+}
+
+#[tokio::test]
+async fn crypto_rsa_pss_sign_verify() {
```

Comment:
I've finally found the reason why this test fails:

First, it fails due to a timeout implemented in the runtime. I increased it locally to debug using `IsolateOptions::tick_timeout` & `IsolateOptions::total_timeout`.

Digging a bit more, I found that `RsaPrivateKey::new_with_exp` took +3s to run, and [looking at the readme](https://github.com/RustCrypto/RSA#example) it's somehow expected. Using an `opt-level` of 3 and the method now takes 150ms:

<img width="858" alt="Screenshot 2023-05-14 at 10 39 09" src="https://github.com/lagonapp/lagon/assets/43268759/6c3bd54f-8302-47c9-98c6-ebbd1a66a719">

Reply:
done